registry/storage/driver/gcs/gcs.go

```diff
@@ -369,18 +384,15 @@ type writer struct {
 	sessionURI string
 	buffer     []byte
 	buffSize   int
+	sClient    *storage.Client
 }
 
 // Cancel removes any written content from this FileWriter.
 func (w *writer) Cancel() error {
 	w.closed = true
-	err := storageDeleteObject(cloud.NewContext(dummyProjectID, w.client), w.bucket, w.name)
-	if err != nil {
-		if status, ok := err.(*googleapi.Error); ok {
-			if status.Code == http.StatusNotFound {
-				err = nil
-			}
-		}
+	err := storageDeleteObject(context.Background(), w.bucket, w.name, w.sClient)
```

Comment:
this is the same thing as what I mentioned earlier. I'm not sure why these funcs are not methods but are instead funcs that accept clients via arg.

Reply:
I think because some of these like `storageDeleteObject` are being used both in writer and driver methods. Writer here, driver elsewhere https://github.com/distribution/distribution/pull/3702/files#diff-1155843a97e1d49ef9bf3a695aa07133afa7e2c94eacb19d0b6052575d41f663R741

Reply:
I think I addressed the other feedback. Let me know how you'd like to proceed here and I'll make the changes.